addons/slack/install.js

```diff
@@ -13,64 +13,97 @@ const getTemplateFiles = (fileNames) =>
   }, {});
 
 module.exports = async (ctx) => {
-  let storageSpecification = await Sdk.createStorage(ctx);
+  if (ctx.caller.permissions) {
+    // Create the Add-On Handler
+    try {
+      // Capture resource path and remove it as we will not need it in the addon instance
+      const eventsResourcePathKey = 'fusebit_events_resource_path';
+      let eventsResourcePath = ctx.body.configuration[eventsResourcePathKey];
+      delete ctx.body.configuration[eventsResourcePathKey];
 
-  // Create the Add-On Handler
-  try {
-    await Sdk.createFunction(ctx, {
-      configurationSerialized: `# Add-on configuration settings
+      await Sdk.createFunction(
+        ctx,
+        {
+          configurationSerialized: `# Add-on configuration settings
 ${Object.keys(ctx.body.configuration)
   .sort()
   .map((k) => `${k}=${ctx.body.configuration[k]}`)
   .join('\n')}
-
-  # Storage configuration settings
-${Object.keys(storageSpecification)
-  .sort()
-  .map((k) => `${k}=${storageSpecification[k]}`)
-  .join('\n')}
 `,
-      nodejs: {
-        files: getTemplateFiles(['index.js', 'package.json', 'slack.js', 'health.js']),
-      },
-      metadata: {
-        fusebit: {
-          editor: {
-            navigationPanel: {
-              hideFiles: ['index.js'],
+          nodejs: {
+            files: getTemplateFiles(['index.js', 'package.json', 'slack.js', 'health.js']),
```

Comment:
Future note: these are the files that should be in the @fuse/slack-client package, with a configuration variable here specifying the name of the client-specific derived npm package to build with.

Reply:
Makes sense, though we have the meeting this afternoon to discuss the exact factoring.